src/vm/i386/gmsx86.cpp

```diff
@@ -1295,10 +1295,10 @@ void LazyMachState::unwindLazyState(LazyMachState* baseState,
     ctx.Esi = lazyState->_esi = baseState->_esi;
     ctx.Ebx = lazyState->_ebx = baseState->_ebx;
 
-    nonVolRegPtrs.Edi = &(lazyState->_edi);
-    nonVolRegPtrs.Esi = &(lazyState->_esi);
-    nonVolRegPtrs.Ebx = &(lazyState->_ebx);
-    nonVolRegPtrs.Ebp = &(lazyState->_ebp);
+    nonVolRegPtrs.Edi = &(baseState->_edi);
```

Comment:
@parjong it looks like you are right that the baseState could be used as the initial place where the context pointers point and that in that case, we would not need to rebase the context pointers. The MachState is also not used after the frame is gone. 
However, since all other architectures do the rebasing, I would prefer following the suite here and doing the same thing for x86.